flake8_type_checking/checker.py

```diff
@@ -399,13 +401,21 @@ def __init__(
         #: Tuple of (node, import name) for all import defined within a type-checking block
         # This lets us identify imports that *are* needed at runtime, for TC004 errors.
         self.type_checking_block_imports: set[tuple[Import, str]] = set()
+
+        #: Set of variable names for all declarations defined within a type-checking block
+        # This lets us avoid false positives for annotations referring to e.g. a TypeAlias
+        # defined within a type checking block
+        self.type_checking_block_declarations: set[str] = set()
+
+        #: Set of all the class names defined within the file
+        # This lets us avoid false positives for classes referring to themselves
         self.class_names: set[str] = set()
 
         #: All type annotations in the file, without quotes around them
-        self.unwrapped_annotations: list[tuple[int, int, str]] = []
+        self.unwrapped_annotations: list[tuple[int, int, str, bool]] = []
```

Comment:
Perhaps it's time to use a dataclass for the contents of the tuple. Something like 

```python
@dataclass()
class Annotation:
    lineno: int 
    col_offset: int
    node_id: str
    is_alias: bool
```

might make it easier to understand what we're storing?

Reply:
I changed it to a NamedTuple in the follow-up PR since that introduced another new field for wrapped annotations.